src/Sentry/Internal/ReleaseLocator.cs

```diff
@@ -4,12 +4,13 @@ namespace Sentry.Internal
 {
     internal static class ReleaseLocator
     {
-        /// <summary>
-        /// Attempts to locate the application release
-        /// </summary>
-        /// <returns>The app release or null, if it couldn't be located.</returns>
-        public static string? GetCurrent()
-            => Environment.GetEnvironmentVariable(Constants.ReleaseEnvironmentVariable)
-                ?? ApplicationVersionLocator.GetCurrent();
+        private static readonly Lazy<string?> CurrentLazy = new(ResolveFromEnvironment);
+
+        private static string? ResolveFromEnvironment() =>
+            Environment.GetEnvironmentVariable(Constants.ReleaseEnvironmentVariable)
+            ?? ApplicationVersionLocator.GetCurrent();
+
+        // Replacing `ResolveFromEnvironment()` with `CurrentLazy.Value` fails tests?
+        public static string? Resolve(SentryOptions options) => options.Release ?? ResolveFromEnvironment();
```

Comment:
It sounds like the state of `Lazy<T>` persists across multiple tests because it's static.